### PR TITLE
require-default-props - fixing an edge case where linting stops early

### PR DESCRIPTION
**Pull request by brgibson**
This is to fix an edge case where linting stops when it finds a component without props.  Instead it should move on to the next component in the file.

### CODE REVIEW

File: tests/lib/rules/require-default-props.js [hunk 1]
```diff
@@ -1306,7 +1306,26 @@ ruleTester.run('require-default-props', rule, {
         column: 3
       }]
     },
-
+    //
+    // component with no pros followed by a failing component
```
Comment (jseminck) [inline tests/lib/rules/require-default-props.js]: Typo.

Also is the empty `//` needed?

File: tests/lib/rules/require-default-props.js [hunk 2]
```diff
@@ -1306,7 +1306,26 @@ ruleTester.run('require-default-props', rule, {
...
+    {
+      code: [
+        'var ComponentWithNoProps = ({ bar = "bar" }) => {',
```
Comment (jseminck) [inline tests/lib/rules/require-default-props.js]: It reads a bit confusing. It says`component with no props` but this component actually has props: `bar` and `foo`. I guess that you actually mean; `component with no declared prop types` (since there is no `.propTypes`) for the first component.

**Comment by jseminck** [comment]
Nice find! 

The fix looks good to me. I added a few comments about the tests.

File: tests/lib/rules/require-default-props.js (hunk 2 above)
Comment (brgibson) [inline tests/lib/rules/require-default-props.js]: updated the comment and fixed typo:

"component with no declared props followed by a failing component"

File: tests/lib/rules/require-default-props.js (hunk 1 above)
Comment (brgibson) [inline tests/lib/rules/require-default-props.js]: I was just following the pattern in the rest of the file, I have deleted it now though, since it does seem unnecessary 👍
